docs/features/analysis.md

```diff
@@ -881,3 +881,40 @@ was greater than `0.90`
 NOTE: if the result is a string, two convenience functions `asInt` and `asFloat` are provided
 to convert a result value to a numeric type so that mathematical comparison operators can be used
 (e.g. >, <, >=, <=).
+## NewRelic Metrics
+
+A [New Relic](https://newrelic.com/) NRQL query can be used to obtain measurements for analysis.  
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: AnalysisTemplate
+metadata:
+  name: success-rate
+spec:
+  args:
+  - name: application-name
+  metrics:
+  - name: success-rate
+    successCondition: result.successRate >= 0.95
+    provider:
+      newrelic:
+        region: "us"
+        accountID: 1234
```

Comment:
I have some questions about the appropriate level of where region and accountID should be configured. i.e. could/should this be configuration at the controller level vs. AnalysisTemplate level? Maybe these clarifying questions would help me better understand:

* can the same account ID be used in different regions? If not, then I think we should eliminate `region` from the analysis spec, and configure the region centrally (same place where the accountID api-key is specified).

* is it common to have multiple accountIDs in the same organization? If most organizations use a single accountID, I would propose to make their experience better / less configuration by not require accountID to be specified in the AnalysisTemplate, and provide a way where a single accountID could be configured centrally. Then AnalysisTemplates would only need to specify `query` and nothing else.

If there is a need / use case to support multiple accountIDs within the same cluster/organization, I think accountID could still be made an **_optional_** field in the analysis spec, but when omitted, the controller could either use designated a "default" accountID / region, or infer the accountID/region/key if there is only a single accountID configured.

Note that a very similar discussion just happened with the Datadog provider:
https://github.com/argoproj/argo-rollouts/pull/705#discussion_r492486540

With DataDog, it was decided that it would be uncommon for a single organization to configure multiple DataDog API/APP keys, so currently argo-rollouts will look to a single Datadog secret for its configuration:

```
apiVersion: v1
kind: Secret
metadata:
   name: datadog
data:
  address: https://api.datadoghq.com
  api-key: igq7imbxlm97vnlw61bo
  app-key: 35bu8ofvc0osyjmqxtgi
```

This doesn't preclude the possibility in the future to support multiple Datadog secrets/api-keys, but it makes the end-user/developer experience better since they did not need to concern themselves with API/APP keys in the AnalysisTemplate.

Another example is with wavefront. It is common to use different wavefront hosts. So, it's configuration looks like this:

Centrally configured Host to API tokens:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: wavefront-api-tokens
type: Opaque
data:
  example1.wavefront.com: <token1>
  example2.wavefront.com: <token2>
```

In the AnalysisTemplate, a reference to **_which_** host is configured:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: AnalysisTemplate
metadata:
  name: success-rate
spec:
  args:
  - name: service-name
  metrics:
  - name: success-rate
    interval: 5m
    successCondition: result >= 0.95
    provider:
      wavefront:
        address: example.wavefront.com
```

Given the DataDog and Wavefront examples, which approach would NewRelic more closely follow?

Reply:
Thanks for the detailed feedback @jessesuen!

Based on your feedback I am working on updating this PR with changes that will more closely follow the pattern used by DataDog. New Relic API keys are specific to an account ID and accounts are specific to a region of New Relic. Given that, the model of putting all of the configuration into the secret to be a global config makes a lot more sense.

I am adding support for multiple secrets out of the gate because we have the need to support multiple accounts within a given cluster. It's fairly common for New Relic users to have primary accounts with sub-accounts that each have their own monitored applications and separate API keys.